packages/sources/oanda/src/transport/modifiedSSE.ts

```diff
@@ -0,0 +1,178 @@
+    return config.SSE_SUBSCRIPTION_TTL
+  }
+
+  override async initialize(
+    dependencies: TransportDependencies<T>,
+    config: AdapterConfig<T['CustomSettings']>,
+    endpointName: string,
+  ): Promise<void> {
+    super.initialize(dependencies, config, endpointName)
+  }
+
+  async streamHandler(
+    context: EndpointContext<T>,
+    subscriptions: SubscriptionDeltas<T['Request']['Params']>,
+  ): Promise<void> {
+    if (subscriptions.new.length || subscriptions.stale.length) {
+      logger.info({ msg: 'Updating SSE subscriptions', subscriptions })
+
+      const subsList = await this.config.parseSubscriptionList(subscriptions.desired, context)
+
+      logger.info({ subsList })
+
+      if (subsList.some((s) => !s)) {
+        logger.error({ msg: `Cannot update SSE subscriptions, subscriptions contains invalid sub` })
+      } else {
+        const sseConfig = this.config.prepareSSEConnectionConfig(subsList as string[], context)
+
+        const providerDataStreamEstablishedUnixMs = Date.now()
+
+        const restartStream = async () => {
+          const response = await axios.get(sseConfig.url, sseConfig.config)
+
+          const stream = response?.data
+
+          const eventHandlerGenerator = (listener: typeof this.config.eventListeners[0]) => {
+            return (event: MessageEvent) => {
+              const providerDataReceivedUnixMs = Date.now()
+
+              const results = listener.parseResponse(event).map((r) => {
+                const partialResponse = r.response as PartialSuccessfulResponse<T['Response']>
+                const result = r as TimestampedProviderResult<T>
+                result.response.timestamps = {
+                  providerDataStreamEstablishedUnixMs,
+                  providerDataReceivedUnixMs,
+                  providerIndicatedTimeUnixMs:
+                    partialResponse.timestamps?.providerIndicatedTimeUnixMs,
+                }
+                return result
+              })
+              this.responseCache.write(results)
+            }
+          }
+
+          const eventHandlers = this.config.eventListeners.reduce(
+            (handlers: { [type: string]: any }, listener) => {
+              handlers[listener.type] = eventHandlerGenerator(listener)
+              return handlers
+            },
+            {},
+          )
+
+          stream.on('open', () => {
+            logger.info({ msg: 'Stream open' })
+          })
+
+          stream.on('close', () => {
+            logger.info({ msg: 'Stream closed, restarting' })
+            restartStream()
+          })
+
+          stream.on('end', () => {
+            logger.info({ msg: 'Stream ended, restarting' })
+            restartStream()
+          })
+
+          stream.on('error', (error: any) => {
+            logger.error({ msg: 'Stream error occurred', error })
+            restartStream()
+          })
+
+          stream.on('data', (data: any) => {
+            logger.error({ data })
```

Comment:
Is this expected to be a logger error?

Reply:
Good catch, I'll change this to `logger.debug`